Description: Try to work around hurd-i386 test failure.
 Per http://polarhome.com/service/man/?qf=unix&tf=2&of=Hurd&sf=7 SO_RCVBUF is not
 supported for AF_UNIX sockets on Hurd. Instead, ENOPROTOOPT is returned when
 we attempt to modify SO_RCVBUF in serialize-async-test.c++.
Author: Tom Lee <debian@tomlee.co>
Bug: https://github.com/sandstorm-io/capnproto/issues/456
Forwarded: no
Last-Update: 2017-04-23
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/capnp/serialize-async-test.c++
+++ b/src/capnp/serialize-async-test.c++
@@ -108,7 +108,10 @@
     // Anyway, we now use 127 to avoid these issues (but also to screw around with non-word-boundary
     // writes).
     uint small = 127;
-    KJ_SYSCALL(setsockopt(fds[0], SOL_SOCKET, SO_RCVBUF, &small, sizeof(small)));
+    if (setsockopt(fds[0], SOL_SOCKET, SO_RCVBUF, &small, sizeof(small)) != 0) {
+        int err = errno;
+        KJ_ASSERT(err == ENOPROTOOPT, "Unable to set SO_RCVBUF for AF_UNIX socket", err);
+    }
     KJ_SYSCALL(setsockopt(fds[1], SOL_SOCKET, SO_SNDBUF, &small, sizeof(small)));
   }
   ~SerializeAsyncTest() {
